flow/connectors/postgres/cdc.go

```diff
@@ -1254,7 +1254,7 @@ func (p *PostgresCDCSource) checkIfUnknownTableInherits(ctx context.Context,
 			if errors.Is(err, pgx.ErrNoRows) {
 				return relID, nil
 			}
-			return 0, fmt.Errorf("failed to query pg_inherits: %w", err)
+			return 0, exceptions.NewPostgresDependencyError(fmt.Errorf("failed to query pg_inherits: %w", err))
```

Comment:
Can you check if there is an underlying connectionc closed in go? Might be simpler

Reply:
I think we can do all 4 of these: https://github.com/jackc/pgx/blob/master/pgconn/pgconn.go#L733-L740